src/main/resources/META-INF/resources/primefaces/datepicker/0-datepicker.js

```diff
@@ -1292,9 +1319,42 @@
             return dayNamesHtml;
         },
 
+        calculateWeekNumber_iso8601: function(d) {
+        	d = new Date(Date.UTC(d.year, d.month, d.day));
+            d.setUTCDate(d.getUTCDate() + 4 - (d.getUTCDay()||7));
+            var yearStart = new Date(Date.UTC(d.getUTCFullYear(),0,1));
+            var weekNo = Math.ceil(( ( (d - yearStart) / 86400000) + 1)/7);
+            return weekNo;
+        },
+
+        calculateWeekNumber_sundayFirst: function (d) {
+            d = new Date(Date.UTC(d.year, d.month, d.day));
+            d.setUTCDate(d.getUTCDate() + 3 - d.getUTCDay());
+            var yearStart = new Date(Date.UTC(d.getUTCFullYear(),0,1));
+            var weekNo = Math.ceil(( ( (d - yearStart) / 86400000) + 1)/7);
+            return weekNo;
+        },
+
+        calculateWeekNumber_saturdayFirst: function (d) {
+        	d = new Date(Date.UTC(d.year, d.month, d.day));
+            d.setUTCDate(d.getUTCDate() - ((d.getUTCDay() + 6) % 7));
+            var yearStart = new Date(Date.UTC(d.getUTCFullYear(),0,1));
+            var weekNo = Math.ceil(( ( (d - yearStart) / 86400000) + 1)/7);
+            return weekNo;
+        },
+
```

Comment:
Feel like there has to be a way to refactor the above 3 methods as there is a lot of duplicate code expecially around yearStart and WeekNo.